Add a Note, GenerateSatelliteAssembliesForCore is related to deterministic option. (#29613)
dotnet/docs: docs/core/extensions/create-satellite-assemblies.md

> </Project>
> ```
>

> > The .NET Core MSBuild task uses *csc.exe* instead of *al.exe* to generate satellite assemblies, by default. When some options, like deterministic option, is used, GenerateSatelliteAssembliesForCore may be needed for effecting satellite assemblies as well. For more information, see [Make it easier to opt into "Core" satellite assembly generation](https://github.com/dotnet/msbuild/pull/2726).

Reviewer: ```suggestion
> The .NET Core MSBuild task uses *csc.exe* instead of *al.exe* to generate satellite assemblies, by default. When some options, for example, the deterministic option are used, `GenerateSatelliteAssembliesForCore` may be needed for effecting satellite assemblies as well. For more information, see [Make it easier to opt into "Core" satellite assembly generation](https://github.com/dotnet/msbuild/pull/2726).
```
